code/go/pkg/validator/validator_test.go

```diff
@@ -290,25 +370,50 @@ func TestValidateFile(t *testing.T) {
 				}
 			}
 
-			if test.expectedErrContains == nil {
+			expectedFailures := 0
+			for _, failedFile := range test {
+				if failedFile.expectedErrContains != nil {
+					expectedFailures += len(failedFile.expectedErrContains)
+				}
+			}
+
+			if expectedFailures == 0 {
 				require.NoError(t, errs)
-			} else {
-				require.Error(t, errs)
-				vErrs, ok := errs.(specerrors.ValidationErrors)
-				if ok {
-					assert.Len(t, errs, len(test.expectedErrContains))
-					var errMessages []string
-					for _, vErr := range vErrs {
+				return
+			}
+
+			require.Error(t, errs)
+
+			vErrs, ok := errs.(specerrors.ValidationErrors)
+			if !ok {
+				require.Len(t, test, 1)
+				require.Len(t, test[0].expectedErrContains, 1)
+				require.Equal(t, errs.Error(), test[0].expectedErrContains[0])
+				return
+			}
+
+			require.Len(t, vErrs, expectedFailures)
+
+			for _, failedFile := range test {
```

Comment:
Thanks for adding support for failures in multiple places, highly appreciated :tada: